packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -236,6 +242,36 @@ _Props/Attrs_
 -   `markdown` - for retrieving and modifying the markdown value. If the client modifies the markdown value, it will be parsed into a Node using the
     [prosemirror-markdown parser](https://github.com/ProseMirror/prosemirror-markdown/blob/9049cd1ec20540d70352f8a3e8736fb0d1f9ce1b/src/from_markdown.ts#L199).
     The parsed node will then be rendered in the viewer component as rich text.
+-   `link-target` - is an attribute that accepts valid `target` values for the `anchor` element, applicable to all links within the viewer component.
+    The following values are accepted by this attribute:
+    1.  `_self`: the current browsing context. (Default)
+    2.  `_blank`: usually a new tab, but users can configure browsers to open a new window instead.
+    3.  `_parent`: the parent browsing context of the current one. If no parent, behaves as `_self`.
+    4.  `_top`: the topmost browsing context (the "highest" context that's an ancestor of the current one). If no ancestors, behaves as `_self`.
+
+Above definitions are from the [MDN reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#:~:text=separated%20link%20types.-,target,-Where%20to%20display).
+
+By default, all the links will open in the same tab as per the accessibility guidelines on WCAG (links are below). If opening in a new tab, `nimble-icon-up-right-from-square` will be placed right
+next to all the links in the viewer instance to indicate the link will be opened in a new tab.
+
+-   Accessibility guidelines to open link only in new tab when required: <https://www.w3.org/TR/WCAG20-TECHS/G200.html>
+-   Accessibility guidelines on opening a link in a new tab: <https://www.w3.org/TR/WCAG20-TECHS/G201.html>
+
+_Alternative implementations:_
+
+For comments feature in SLE, there is a requirement to have all links open in a new tab. However, we are also considering an alternative approach where links will open in a new tab only if
+they are external (i.e., not from the same domain/origin); for internal links, they will open in the same tab. Here, we will show the same warning mentioned above for the external
+links. Below is proposed high level design for the same,
+
+We will introduce an attribute named `link-configuration` that can accept the following enum values:
+
+1.  `openInSameTab` - This value is the default setting. If chosen, all links within the component will open in the same tab.
+2.  `openInNewTab` - When selected, all links in the component will open in new tab. A warning icon will be placed next to each link, conveying the user that
+    clicking the link will result in a new tab opening.
+3.  `externalLinkBasedOnOrigin` - With this setting, links will be validated based on their origins. If a link's origin differs from the current origin, it is categorized as
+    external link and will open in a new tab. Conversely, links originating from the same domain will be treated as internal link and open in the same tab.
```

Comment:
I don't think this alternate design is sufficient to be a core features in nimble.

I believe the concern generally is that if a URL is trusted or not-trusted and to warn the user about untrusted URLs.

Just because a URL is a different origin does not mean it is untrusted and should have a warning.
For example, a URL on ni.com would not show a warning for a URL to systemlink.ni.com or for a url to systemlinkcloud.com. Those URLs are different origins, but they are not untrusted URLs.

We may need additional discussion but as far as I can tell the definition of an "untrusted URL" is application specific. Just having a different origin is not a sufficient indication that a URL is not trustworthy.

> If a link's origin differs from the current origin, it is categorized as external link and will open in a new tab

Based on the WCAG guidelines this is not a sufficient reason to always open a URL in a new tab. If the URL is untrusted we need a way for the user to visualize that the URL is untrusted but we do not have that visual yet and that visual is not a requirement yet because we only currently support absolute URLs.

Reply:
This is a valid concern on getting to a right definition for the trusted or untrusted links specific to application. 

This alternative approach is not applicable now since the current working decision is to go with the default behavior of same tab. Contents has been removed from this HLD